pkg/azure/blob.go

```diff
@@ -180,14 +181,24 @@ func (c *BlobClient) BucketExists(ctx context.Context, bucketName string) (bool,
 	if err != nil {
 		return false, err
 	}
-	_, err = container.GetProperties(ctx, nil)
-	if err != nil {
-		var stgErr *azblob.StorageError
-		if errors.As(err, &stgErr) {
-			if stgErr.ErrorCode == azblob.StorageErrorCodeContainerNotFound {
+
+	items := container.ListBlobsFlat(&azblob.ContainerListBlobsFlatOptions{
+		MaxResults: to.Ptr(int32(1)),
+	})
+	// We call next page only once since we just want to see if we get an error
+	items.NextPage(ctx)
+	if err := items.Err(); err != nil {
+		var respErr *azcore.ResponseError
+		if errors.As(err, &respErr) {
+			if respErr.ErrorCode == string(*azblob.StorageErrorCodeContainerNotFound.ToPtr()) {
 				return false, nil
 			}
-			err = stgErr
+			err = respErr
+
+			// For a container-level SASToken, we get an AuthenticationFailed when the bucket doesn't exist
+			if respErr.ErrorCode == string(azblob.StorageErrorCodeAuthenticationFailed) {
```

Comment:
The library seems to have a helper method for this since `v0.5.0`: https://github.com/Azure/azure-sdk-for-go/commit/0e6a11eae55a10b7f2b0f5032afc80d932642a81#diff-2c126bccb5ce0d58738ebd81c556270990a2f9f71a5bbfe8021f458c919c72aaR18

Reply:
The upgrade to `v0.5.0` introduces some other changes that I think should come in a different PR.
Wdyt?